course listing

CalebLink18 opened the pull request:
I had to manually add every file I have created/altered to a new branch to get this to work, just as a heads up.

CalebLink18 committed:
Add files via upload

CalebLink18 committed:
take 2

CalebLink18 committed:
Delete champlain-api-courses2 directory

was inserted on accident

CalebLink18 committed:
Add files via upload

CalebLink18 committed:
Add files via upload

CalebLink18 committed:
Add files via upload

CalebLink18 committed:
Add files via upload

jnord0 commented on api/prisma/seed.ts:
```diff
@@ -62,6 +62,46 @@ async function addSeedData() {
+        }
+    }
+
+    console.log("✅ Database seeding completed!");
```
Could you delete the console log

CalebLink18 replied:
done

CalebLink18 commented:
sure, gimme a moment


On Thu, Apr 3, 2025 at 2:02 PM Jack Nordberg ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In api/prisma/seed.ts
> <https://github.com/champlain-api/champlain-api/pull/53#discussion_r2027503659>
> :
>
> > +                    times: courseData.times,
> +                    instructor_name: courseData.instructor_name,
> +                    description: courseData.description,
> +                    room: courseData.room,
> +                    subject: courseData.subject,
> +                    type: courseData.type,
> +                    prereq: courseData.prereq,
> +                    start_date: courseData.start_date, // Keeping as string
> +                    end_date: courseData.end_date, // Keeping as string
> +                    semesterId: semester.id, // Linking course to semester
> +                }
> +            });
> +        }
> +    }
> +
> +    console.log("✅ Database seeding completed!");
>
> Could you delete the console log
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/champlain-api/champlain-api/pull/53#pullrequestreview-2740736212>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BCIVCXRI5XUMLP3WZVCE33T2XVZTFAVCNFSM6AAAAAB2KBVI3WVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDONBQG4ZTMMRRGI>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>

CalebLink18 committed:
Update seed.ts

removed a console log

hayden36 commented on api/src/routes/courses2.ts:
```diff
@@ -0,0 +1,223 @@
+/*
+courses2.ts because the first one was a failure
+
+This defines the routes for the courses endpoint.`
+
+Users can do the following:
+
+Request all courses
+ *  GET /courses
+ *      Returns a 500 if an error occurs.
+ * 
+Request a specific semesters worth of courses
+ * GET /courses/:semester
+ *     (incomplete)
+ *     Returns a 500 if an error occurs.
+ * 
+Request a specific course
+ * GET /courses/:courseNumber
+ *     (incomplete)
+ * 
+Post a specific course
+ * POST /courses
+ *     (incomplete)
+
+*/
+
```
Can you update this comment and rename the file?

CalebLink18 replied:
in my branch now

hayden36 commented on api/src/routes/courses2.ts:
```diff
@@ -0,0 +1,223 @@
+/**
+ * @route GET /courses/:semester
+ * @description Fetch all courses of a specific semester
+ * @param {string} semesterId - The ID of the semester to fetch courses for. * @returns {JSON} List of courses.
+ * @returns {JSON} List of courses. 
```
The `@returns` is here twice.

CalebLink18 replied:
done in my branch now

hayden36 commented on api/src/routes/courses2.ts:
```diff
@@ -0,0 +1,223 @@
+    .get("/:semesterId", async (req: Request, res: Response) => {
+        res.setHeader("Content-Type", "application/json");
+
+        const semesterId = parseInt(req.params.semesterId, 10);
```
I feel like it's unlikely that anyone is going to give the semester ID as anything other than base 10.

CalebLink18 replied:
I have removed that now.

hayden36 commented on api/src/data/semesters_and_courses.json:
Can we only have a couple courses in here? I don't think its necessary to give this big of a file. I'll make a PR for when I finish my tests to do the same to the faculty file.

CalebLink18 replied:
ok, i am deleting the json file and just gonna have a couple example semesters and classes.

hayden36 commented on api/prisma/migrations/20250305171136_/migration.sql:
There should be a new migration for this endpoint.

CalebLink18 replied:
there is one?

hayden36 replied:
I meant a new file. it shouldn't be editing the initial migration.

CalebLink18 replied:
i see.
ok, done.

hayden36 commented on api/prisma/schema.prisma:
```diff
@@ -66,4 +66,33 @@ enum APIKeyScopes {
 }
+
+model Course {
+  id              Int      @id @default(autoincrement())
```
Wouldn't it make more sense to have the ID be like `CSI-318` instead of an integer?

hayden36 commented on api/prisma/schema.prisma:
```diff
@@ -66,4 +66,33 @@ enum APIKeyScopes {
+model Course {
+  id              Int      @id @default(autoincrement())
+  title           String   @default("")
+  number          String   @default("")
```
Is this supposed to be a section?

hayden36 commented on api/src/routes/courses2.ts:
Having the two `GET`s will conflict I'm pretty sure.

CalebLink18 committed:
Add files via upload

CalebLink18 committed:
Add files via upload

CalebLink18 committed:
Delete api/src/data/semesters_and_courses.json

data is inputted in seed.ts now

CalebLink18 committed:
Add files via upload

first four issues with pull are now done

CalebLink18 committed:
Add files via upload

CalebLink18 committed:
Add files via upload

finished revisions

CalebLink18 committed:
Add files via upload

CalebLink18 committed:
Delete api/src/data directory

no longer needed

CalebLink18 committed:
Add files via upload

closer to up to date with main

CalebLink18 committed:
Delete api/prisma/migrations/20250305171136_ directory

no longer needed

hayden36 commented on api/src/routes/courses2.ts:
Can you delete this file if it's not being used?

CalebLink18 replied:
my bad. done

hayden36 commented on api/src/routes/courses.ts:
Can you update the comments in the beginning of the file and add a `PUT` and a `DELETE` option?

CalebLink18 committed:
Delete api/src/routes/courses2.ts

the name was changed

CalebLink18 committed:
Add files via upload

changed some comments

CalebLink18 committed:
Add files via upload

set up a better migration file

CalebLink18 committed:
Merge branch 'main' into courses2

loganc04 commented:
@CalebLink18 do you think you can get this finished by the end of this week?

CalebLink18 commented:
yes, that is the plan. pull and delete should be the last things.

On Wed, Apr 16, 2025 at 8:08 PM Logan Clark ***@***.***>
wrote:

> @CalebLink18 <https://github.com/CalebLink18> do you think you can get
> this finished by the end of this week?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/champlain-api/champlain-api/pull/53#issuecomment-2811276761>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BCIVCXUXRAST7VAT3YHIJWT2Z3WHXAVCNFSM6AAAAAB2KBVI3WVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQMJRGI3TMNZWGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
> *loganc04* left a comment (champlain-api/champlain-api#53)
> <https://github.com/champlain-api/champlain-api/pull/53#issuecomment-2811276761>
>
> @CalebLink18 <https://github.com/CalebLink18> do you think you can get
> this finished by the end of this week?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/champlain-api/champlain-api/pull/53#issuecomment-2811276761>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BCIVCXUXRAST7VAT3YHIJWT2Z3WHXAVCNFSM6AAAAAB2KBVI3WVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQMJRGI3TMNZWGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

CalebLink18 committed:
Add files via upload

fixed an admin error

CalebLink18 committed:
Add files via upload

fixed spikeyscopes to include the course and semester tables

CalebLink18 committed:
Add files via upload

update

CalebLink18 committed:
Add files via upload

test update

CalebLink18 committed:
Add files via upload

CalebLink18 committed:
Delete api/prisma/migrations/20250416120000_courses directory

test

hayden36 commented:
If you have bun installed, you can run `bun test` to run the tests. you might have to run `bunx prisma migrate reset -f` to reset the database before the tests run,

CalebLink18 committed:
Add files via upload

undo test

CalebLink18 committed:
a correct migraion file

CalebLink18 committed:
Delete 20250417232242_add_courses directory

cannot move this simply

CalebLink18 committed:
migration fix

CalebLink18 committed:
less stupid looking seed.ts

CalebLink18 committed:
Delete api/prisma/migrations/20250416120000_courses directory

should not exist. was a failure at life. will not be missed

CalebLink18 committed:
added update and delete options

when playing with prisma remember to drink all the red cool aid. for she will not let you do shit untill you do.

hayden36 commented on api/prisma/schema.prisma:
```diff
@@ -96,17 +69,58 @@ model Housing {
   COMPETENCIES_EDIT
+  BUILDING_EDIT
+  COURSES_EDIT
+  SEMESTER_EDIT
```
Looks like `SEMESTER_EDIT` isn't used.

hayden36 commented on api/src/routes/courses.ts:
```diff
@@ -0,0 +1,304 @@
+ * @description Adds a new course to a specific semester
+ * @returns {JSON} The created course.
+ */
+    .post("/courses", requireAPIKeyScopes([APIKeyScopes.COURSES_EDIT]), async (req: Request, res: Response) => {
```
I think this should be `post("/")`. You're defining this route as `/course` in the index file.

CalebLink18 replied:
done

hayden36 commented on api/src/routes/courses.ts:
```diff
@@ -0,0 +1,304 @@
+ * @param {number} courseId - The ID of the course to update.
+ * @returns {JSON} The updated course.
+ */
+router.put("/courses/:courseId", requireAPIKeyScopes([APIKeyScopes.COURSES_EDIT]), async (req: Request, res: Response) => {
```
Same here. I think it should be `put("/:courseId")`

CalebLink18 replied:
done

hayden36 commented on api/src/routes/courses.ts:
It doesn't look like there's an ability to create semesters. Can you add this?

CalebLink18 replied:
done

hayden36 commented on api/src/routes/courses.ts:
Can you also add the license header that's present in the other files too?

hayden36 reviewed (changes requested):
Looks good so far, just a couple changes. Can you also add tests to make sure that everything is working correctly?

CalebLink18 commented:
interesting. 

what is the point of creating semesters and worrying about the difference between "/courses/:courseId" and "/:courseId" since you know that this is going to get depricated in less than a year? you told me that all we were going for here was an example of the courses and semesters. 

unless you plan to take your personal project further and add a function for that in the future there is no point to doing that.

CalebLink18 committed:
headers update

hayden36 commented:
Right now there's no way to add courses unless you add a semester manually in the database.


> difference between "/courses/:courseId" and "/:courseId" 

If someone wants to edit a course, they have to make a request to `/course/courses/CSI-240`, for example. The /course is there twice.

CalebLink18 committed:
add course test files

CalebLink18 committed:
added semester control

CalebLink18 committed:
course tests update

CalebLink18 committed:
course tests update 2

added sesester routes updates

CalebLink18 committed:
course tests update 3

is now reliant on mock data

CalebLink18 committed:
tests fix

CalebLink18 committed:
tests fix2

CalebLink18 committed:
remove tests

CalebLink18 committed:
Delete api/src/tests/courses.test.ts

CalebLink18 committed:
Delete api/src/types/courses.d.ts

CalebLink18 commented:
I do not have time to work on something that was never originally necessary. I will not add the tests and as of now this should be done.

hayden36 commented:
I think you accidentally uploaded the prisma and src directory to the root of the project.

loganc04 commented:
> I think you accidentally uploaded the prisma and src directory to the root of the project. Also tests have been an issue for a couple weeks and we've all been able to add them. The tests are there so we know the endpoints are working as expected.

Yeah, we added tests in #55 which was two weeks ago.

CalebLink18 committed:
Delete prisma directory

CalebLink18 committed:
Delete src directory

CalebLink18 commented:
go ahead and merge.

hayden36 commented:
The `SEMESTER_EDIT` key scope is still unused.

hayden36 commented on api/src/routes/courses.ts:
```diff
@@ -0,0 +1,496 @@
+ * @description Returns all existing semesters
+ * @returns {JSON} A list of semesters or an error.
+ */
+router.get("/semester", async (req: Request, res: Response) => {
```
This doesn't work because of the `GET /:courseNumber` above, but i'll approve it now for the sake of time.

hayden36 commented:
The `GET /semster` doesn't work because of the GET /:courseNumber above, but i'll approve it now for the sake of time.
